packages/components/src/Button/ButtonCopyToClipboard.tsx

```diff
@@ -0,0 +1,103 @@
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import React, { useRef, useState, FC, cloneElement } from 'react'
+import styled from 'styled-components'
+import { ButtonOutline } from './ButtonOutline'
+
+/**
+ * This component allows user to copy contents from the, passed in ref, to the clipboard.
+ */
+
+export interface CopyToClipboardProps {
+  /**
+   * Content to be copied into the clipboard
+   */
+  content: string
+  /**
+   * button's label | a JSX element to replace the button
+   * @default 'Copy to Clipboard'
+   */
+  children?: string | JSX.Element
+  /**
+   * button's successfully copied label | a JSX element to replace the button
+   * @default 'Copied!'
+   */
+  success?: string | JSX.Element
+}
+
+const SuccessButton = styled(ButtonOutline)`
+  pointer-events: none;
+`
+
+export const CopyToClipboard: FC<CopyToClipboardProps> = ({
+  children = 'Copy to Clipboard',
+  content,
+  success = 'Copied',
+}) => {
+  const [copied, setCopied] = useState(false)
+  const buttonRef = useRef<HTMLButtonElement>(null)
+
+  const clickCopyButton = () => {
+    const textField = document.createElement('textarea')
+    textField.value = content
+
+    if (buttonRef.current) {
+      buttonRef.current.appendChild(textField)
+      textField.select()
+      document.execCommand('copy')
+      textField.remove()
+      setCopied(true)
+      setTimeout(() => setCopied(false), 2500)
+    }
+  }
+
+  const copyButton =
+    typeof children === 'string' ? (
+      <ButtonOutline
+        iconBefore="Clipboard"
+        onClick={clickCopyButton}
+        ref={buttonRef}
+      >
+        {children}
+      </ButtonOutline>
+    ) : (
+      cloneElement(children, {
+        onClick: clickCopyButton,
+        ref: buttonRef,
+      })
+    )
+
+  const successButton =
+    typeof success === 'string' ? (
+      <SuccessButton iconBefore="Check">{success}</SuccessButton>
+    ) : (
+      cloneElement(success, {
+        style: { pointerEvents: 'none' },
+      })
```

Comment:
I think `pointerEvents: 'none'` will cause accessibility and focus issues - simply not having an `onClick` should prevent any user actions while the button is in it's success state.

```suggestion
  success
```

Reply:
Some of the pointer events changes might be because I gave some feedback that the success state of the button feels strange that it's still clickable.  It might not be that big of a deal in the short term until we can come up with a more holistic approach.

I tested it with voice over, and in either case, with pointer events or not, voice over always puts focus onto the example `textarea` and never put back on the `Button`.  I checked `document.activeElement` after clicking the button and it states that the `<body>` element has focus.

We will probably also need `aria-live="polite"` on the success button here so that we can announce that the copy was successful.

Reply:
Ok I remove the `pointerEvents: 'none'` and add `aria-live="polite"` instead. I haven't learned how to test the accessibility yet. But from what i see if we don't have `pointerEvents: 'none'` the success button will be clickable.